tests/upgrade/0.8.2-0.9.0/jiva/suite_test.go

```diff
@@ -0,0 +1,222 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	// auth plugins
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+var (
+	kubeConfigPath        string
+	replicaCount          int
+	nsName                = "default"
+	scName                = "jiva-upgrade-sc"
+	openebsProvisioner    = "openebs.io/provisioner-iscsi"
+	replicaLabel          = "openebs.io/replica=jiva-replica"
+	ctrlLabel             = "openebs.io/controller=jiva-controller"
+	openebsCASConfigValue = "- name: ReplicaCount\n  Value: "
+	accessModes           = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+	capacity              = "5G"
+	pvcObj                *corev1.PersistentVolumeClaim
+	pvcName               = "jiva-volume-claim"
+	scObj                 *storagev1.StorageClass
+	openebsURL            = "https://openebs.github.io/charts/openebs-operator-0.8.2.yaml"
+	rbacURL               = "https://raw.githubusercontent.com/openebs/openebs/master/k8s/upgrades/0.8.2-0.9.0/rbac.yaml"
+	crURL                 = "https://raw.githubusercontent.com/openebs/openebs/master/k8s/upgrades/0.8.2-0.9.0/jiva/cr.yaml"
+	runtaskURL            = "https://raw.githubusercontent.com/openebs/openebs/master/k8s/upgrades/0.8.2-0.9.0/jiva/jiva_upgrade_runtask.yaml"
+	jobURL                = "https://raw.githubusercontent.com/openebs/openebs/master/k8s/upgrades/0.8.2-0.9.0/jiva/volume-upgrade-job.yaml"
+)
+
+func TestSource(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test jiva volume upgrade")
+}
+
+func init() {
+	flag.StringVar(&kubeConfigPath, "kubeconfig", "", "path to kubeconfig to invoke kubernetes API calls")
+	flag.IntVar(&replicaCount, "replicas", 1, "number of replicas to be created")
+}
+
+var ops *tests.Operations
+
+var _ = BeforeSuite(func() {
+
+	ops = tests.NewOperations(tests.WithKubeConfigPath(kubeConfigPath))
+	openebsCASConfigValue = openebsCASConfigValue + strconv.Itoa(replicaCount)
+
+	// Setting the path in environemnt variable
+	err := os.Setenv(string(v1alpha1.KubeConfigEnvironmentKey), kubeConfigPath)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	By("applying openebs 0.8.2")
+	applyFromURL(openebsURL)
+
+	By("waiting for maya-apiserver pod to come into running state")
+	podCount := ops.GetPodRunningCountEventually(
+		string(artifacts.OpenebsNamespace),
+		string(artifacts.MayaAPIServerLabelSelector),
+		1,
+	)
+	Expect(podCount).To(Equal(1))
+
+	annotations := map[string]string{
+		string(apis.CASTypeKey):   string(apis.JivaVolume),
+		string(apis.CASConfigKey): openebsCASConfigValue,
+	}
+
+	By("building a storageclass")
+	scObj, err = sc.NewBuilder().
+		WithName(scName).
+		WithAnnotations(annotations).
+		WithProvisioner(openebsProvisioner).Build()
+	Expect(err).ShouldNot(HaveOccurred(), "while building storageclass {%s}", scName)
+
+	By("creating above storageclass")
+	_, err = ops.SCClient.Create(scObj)
+	Expect(err).To(BeNil(), "while creating storageclass {%s}", scObj.Name)
+
+	By("building a pvc")
```

Comment:
In near future, we need to check if current folder structure and logic in this `suite_test.go` is good enough to test scenarios like presence of multiple old volumes & upgrading only one volume & similar combinations.